be/src/runtime/routine_load/kafka_consumer_pipe.h

```diff
@@ -50,7 +50,11 @@ class KafkaConsumerPipe : public StreamLoadPipe {
         return st;
     }
 
-    Status append_json(const char* data, size_t size, char row_delimiter) { return append_and_flush(data, size); }
+    Status append_json(const char* data, size_t size, char row_delimiter) {
+        auto buf = ByteBuffer::allocate(size);
```

Comment:
```suggestion
        auto buf = ByteBuffer::allocate(size + simdjson::SIMDJSON_PADDING);
```

Reply:
> Yes. But if it is done by the producer of `stream_load_pipe`, since there're many producers, it would make the code fragmented. It's not easy to remind every programmer to remember this. Besides, the `simdjson::SIMDJSON_PADDING` only cost several bytes. There's usually extra space in the `ByteBuffer`, so the possibility of re-allocation is low.

This should be considered.

Reply:
IMO, It's ok to leave the checks in `json_scanner.cpp` untouched, but you can still allocate some extra space here for better performance otherwise, there will always be an extra memory allocation and memory copy in json_scanner.cpp